phases/option.go

```diff
@@ -0,0 +1,72 @@
+package phases
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type option struct {
+	title        string
+	valueOptions []string
+	action       func(string) *option
+}
+
+func (o *option) run() {
+	answer := ask(o.title, o.valueOptions...)
+	if o.action != nil {
+		if nextOption := o.action(answer); nextOption != nil {
+			nextOption.run()
+		}
+	}
+}
+
+func ask(title string, options ...string) string {
+	if len(options) == 1 {
+		return options[0]
+	}
+
+	fmt.Print(strings.TrimSuffix(title, ":") + ":")
+
+	if len(options) == 0 {
+		fmt.Print(" ")
+		for {
+			input, err := bufio.NewReader(os.Stdin).ReadString('\n')
+			if err != nil {
+				log.Errorf("Error: failed to read input value")
+				continue
+			}
+			fmt.Println()
+			return strings.TrimSpace(input)
+		}
+	}
+
+	fmt.Println()
+	for i, option := range options {
+		log.Printf("(%d) %s", i+1, option)
+	}
+
+	for {
+		fmt.Print("Option number: ")
+		answer, err := bufio.NewReader(os.Stdin).ReadString('\n')
+		if err != nil {
+			log.Errorf("Error: failed to read input value")
```

Comment:
NOTE: i would not include the `Error: ` prefix, since log coloring and the message clearly shows that there was an issue.

PLS FIX: include the error in the message, to let the user debug it and fix it.